envoy/network/dns_resolver.h

```diff
@@ -40,16 +40,20 @@ class DnsResolverFactory : public Config::TypedFactory {
    * needs to be cleaned up before terminates Envoy.
    */
   virtual void terminate() {}
-};
 
-/**
- * Terminate the DNS resolver factories.
- */
-static inline void terminateDnsResolverFactories() {
-  auto& factories = Registry::FactoryRegistry<Network::DnsResolverFactory>::factories();
-  std::for_each(factories.begin(), factories.end(),
-                [](auto& factory_it) { factory_it.second->terminate(); });
-}
+  /**
+   * Find the DNS resolver factory based on the typed config and initialize it.
+   * @returns the DNS Resolver factory.
+   * @param typed_dns_resolver_config: the typed DNS resolver config
+   */
+  static Network::DnsResolverFactory& initializeDnsResolverFactory(
```

Comment:
great! Now that we are inside DnsResolverFactory I think we can rename this method to createFactory(), and we can rename terminateDnsResolverFactories to terminateFactory().

The `DnsResolver` part of the names now look redundant at call-sites.